runtime/interpreter/value.go

```diff
@@ -664,39 +723,50 @@ func (v *ArrayValue) Append(element Value) {
 	v.modified = true
 
 	element.SetOwner(v.Owner)
-	v.Values = append(v.Values, element)
+
+	//nolint:gocritic
+	v.values = append(v.Elements(), element)
```

Comment:
Maybe just use `ensureElementsLoaded` before the `append`, like in `SetIndex`?
Same below in `AppendAll`

```suggestion
	v.ensureElementsLoaded()
	v.values = append(v.values, element)
```